### PR TITLE
install command: add cache-only option

### PR DESCRIPTION
**Pull request by becker33**
Creates the option to install only from binary cache, ignoring Spack's ability to build from source.

### CODE REVIEW

File: lib/spack/spack/package.py [hunk 1]
```diff
@@ -1786,7 +1789,6 @@ def check_for_unfinished_installation(
 
         if restage and self.stage.managed_by_spack:
             self.stage.destroy()
-            self.stage.create()
```
Comment (adamjstewart) [inline lib/spack/spack/package.py]: Why is this line removed?

**Comment by adamjstewart** [comment]
Also need to update Bash completion script

File: lib/spack/spack/cmd/bootstrap.py [hunk 2]
```diff
@@ -34,6 +34,9 @@ def setup_parser(subparser):
     cache_group.add_argument(
         '--no-cache', action='store_false', dest='use_cache', default=True,
         help="do not check for pre-built Spack packages in mirrors")
+    cache_group.add_argument(
+        '--cache-only', action='store_true', dest='cache_only', default=False,
```
Comment (alalazo) [inline lib/spack/spack/cmd/bootstrap.py]: Just a comment, not a request for any change. Using `store_true` as an action already sets the default to `False`. Vice-versa is valid for `store_false`. See [here](https://docs.python.org/2/library/argparse.html#action) for reference. Also setting `dest=cache_only` should be unnecessary, in the sense that coincides with the default given the argument name.

File: lib/spack/spack/package.py (hunk 1 above)
Comment (becker33) [reply lib/spack/spack/package.py]: Because the stage destruction occurs before the stage should be created. Failed installations using `spack install --cache-only` were creating a stage directory at this point, and there's no reason to create the directory yet. We don't even know yet whether we need it. So it's better to only destroy it now, and recreate it when we know whether we need it.
